pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/RoundDecimalTransformFunction.java

```diff
@@ -90,8 +90,12 @@ public double[] transformToDoubleValuesSV(ValueBlock valueBlock) {
     double[] leftValues = _leftTransformFunction.transformToDoubleValuesSV(valueBlock);
     if (_fixedScale) {
       for (int i = 0; i < length; i++) {
-        _doubleValuesSV[i] = BigDecimal.valueOf(leftValues[i])
-            .setScale(_scale, RoundingMode.HALF_UP).doubleValue();
+        if (Double.NEGATIVE_INFINITY == leftValues[i]) {
+          _doubleValuesSV[i] = Double.NaN;
```

Comment:
Can you check the standard SQL behavior when rounding on `-Inf`? Does it throw exception, return `-Inf` or `NaN`?
We should put the same check in other if branches

Reply:
I would take a look.

Reply:
It is DB specific behaviour. Some throw errors, some result to -infinity.

Reply:
How about PostgreSQL? We usually use it as a reference

Reply:
PostgreSQL returns -Infinity.

Reply:
Let's try to match that behavior. We want to handle all special values, including `Inf`, `-Inf`, `NaN`

Reply:
But I undrstand that currently pinot stores null values as -Inf only. Why do we need ot handle other two?

Reply:
Which layer eventually writes to response to a SQL query?

Reply:
Pinot does use `-Inf` as the placeholder for null value, but `Inf` is also valid value and we should be able to handle it without throwing exception. `NaN` is debatable because they are converted into `null` within `SpecialValueTransformer`